tests/backup/backup_orphaned_delete_test.go

```diff
@@ -0,0 +1,505 @@
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			createBackup := func(backupName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				for backupLocationUID, backupLocationName := range userBackupLocationMap {
+					_, err = CreateBackupByNamespacesWithoutCheck(backupName, SourceClusterName, backupLocationName, backupLocationUID,
+						[]string{invalidNamespace}, map[string]string{}, orgID, userSourceClusterUID, "", "", "", "", nonAdminCtx)
+					if err != nil {
+						backupUID, UIDError := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+						log.FailOnError(UIDError, "failed to get backup %s uid", backupName)
+						backupInspectRequest := &api.BackupInspectRequest{
+							Name:  backupName,
+							OrgId: orgID,
+							Uid:   backupUID,
+						}
+						resp, inspectError := Inst().Backup.InspectBackup(nonAdminCtx, backupInspectRequest)
+						log.FailOnError(inspectError, "failed to inspect backup %s", backupName)
+						if resp.GetBackup().GetStatus().GetStatus() != api.BackupInfo_StatusInfo_Failed {
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup %s of namespace %s", backupName, namespace))
+						}
+					}
+					break
+				}
+			}
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+					userFailedBackupNames = append(userFailedBackupNames, backupName)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of failed user backups taken are: %v", userFailedBackupNames)
+		})
+		Step(fmt.Sprintf("Verify failed backups of the user %s from the admin", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Verifying failed backups of the user %s from the admin", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			userOwnerID, err := portworx.GetSubFromCtx(nonAdminCtx)
+			log.FailOnError(err, "failed to fetch user owner id %s", infraAdminUser)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupNamesByOwnerID, err := GetAllBackupNamesByOwnerID(userOwnerID, orgID, ctx)
+			log.FailOnError(err, "failed to fetch backup names with owner id %s from the admin", userOwnerID)
+			for _, backupName := range userFailedBackupNames {
+				if !IsPresent(backupNamesByOwnerID, backupName) {
+					err := fmt.Errorf("failed backup %s is not listed in backup names %s", backupName, backupNamesByOwnerID)
+					log.FailOnError(fmt.Errorf(""), err.Error())
+				}
+			}
+		})
+		Step(fmt.Sprintf("Delete failed backups taken by user %s from the admin", infraAdminUser), func() {
```

Comment:
Can we combine this and the previous step i.e check if the backup is present -> if yes -> delete

Reply:
In TestRail these are two different steps.